### PR TITLE
feat: add health check

### PR DESCRIPTION
**Pull request by kokoraka**
Changes:
- add logging package
- add health check package
- add rest app package
- upadte serialization
- generate mock

Evidence:
1. Not found handler
image

2. Method not allowed handler
image

3. Root endpoint
image

4. Health check endpoint
- All checks were failed
image

- Some checks were failed
image

### CODE REVIEW

File: internal/healthcheck/go_health.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	}
+
+	return s.Client.Start()
+}
+
+func (s *GoHealthService) Stop() error {
+	return s.Client.Stop()
+}
+
+func (s *GoHealthService) Check() (*CheckResult, error) {
```
Comment (ahmadqosam-photon) [inline internal/healthcheck/go_health.go:39]: I find myself do not understand "why" this function exists as well as other methods. For new joiner like me, it is not clear what to achieve with these methods. Minutes are taken to understand the whole idea.

Do we need to add comments that explain these things like some open source projects?
- Why (problem),
- what to achieve.

Comment (yudium) [inline internal/healthcheck/go_health.go:39]: I find myself do not understand "why" this function exists as well as other methods. Perhaps these are source of my confusion :

- I am still new to health check things
- The codes can improved by documentation? such as comment to explain "why" and "goal" that common in open source projects. But hope we do not need many comments as long as we able to understand code more faster.

File: internal/rest-app/handler.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	"github.com/go-seidon/local/internal/logging"
+	"github.com/go-seidon/local/internal/serialization"
+)
+
+func NewNotFoundHandler(log logging.Logger, serializer serialization.Serializer) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		log.Debug("In function: NotFoundHandler")
+		defer log.Debug("Returning function: NotFoundHandler")
+
+		w.Header().Set("Content-Type", "applicaton/json")
```
Comment (yudium) [inline internal/rest-app/handler.go:17]: typo missing i char: applic**ato**n

Comment (kokoraka) [reply internal/rest-app/handler.go:17]: Nice catch!

File: internal/healthcheck/go_health.go (hunk 1 above)
Comment (kokoraka) [reply internal/healthcheck/go_health.go:39]: Health check: checking current services condition such as their connectivity within other service, connectivity to the internal, disk usage, memory usage, etc

**Pull request merged**